File: benchmarks/lua/records.lua
local n = tonumber(arg[1] or "300000")
local points = {}
for i = 0, n - 1 do
    points[#points + 1] = {x = i % 1000, y = i % 7}
end
local total = 0
for i = 1, #points do
    local p = points[i]
    total = total + p.x + p.y
end
print(total)

File: benchmarks/python/records.py
import sys

class Point:
    __slots__ = ("x", "y")

    def __init__(self, x, y):
        self.x = x
        self.y = y

def main():
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 300000
    points = []
    for i in range(n):
        points.append(Point(i % 1000, i % 7))
    total = 0
    for i in range(len(points)):
        p = points[i]
        total = total + p.x + p.y
    print(total)

main()
